src/boxplot.rs

```diff
@@ -0,0 +1,241 @@
+use super::{AsMatrix, matrix_to_array, generate_list, GraphMaker};
+use std::fmt::Write;
+
+/// Draw a box and whisker plot
+///
+/// [See Matplotlib's documentation](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.boxplot.html)
+///
+/// # Examples
+///
+/// ## Basic box plot
+///
+/// ```
+/// use plotpy::{Boxplot, AsMatrix, Plot, StrError};
+///
+/// fn main() -> Result<(), StrError> {
+///     // data
+///     let x = vec![vec![1,2,3,4,5],
+///                  vec![2,3,4,5,6],
+///                  vec![3,4,5,6,7],
+///                  vec![4,5,6,7,8],
+///                  vec![5,6,7,8,9],
+///                  vec![6,7,8,9,10]];
+/// 
+///     // x ticks and labels
+///     let ticks: Vec<_> = (1..=x.size().1).into_iter().collect();
+///     let labels = ["x1", "x2", "x3", "x4", "x5"];
+/// 
+///     // boxplot object and options
+///     let mut boxes = Boxplot::new();
+///     boxes.set_vertical(true)
+///          .draw(&x);
+///
+///     // save figure
+///     let mut plot = Plot::new();
+///     plot.add(&boxes).set_title("boxplot")
+///         .set_ticks_x_labels(&ticks, &labels)
+///         .set_labels("x", "y")  // x-axis and y-axis label
+///         .save("/tmp/plotpy/doc_tests/doc_boxplot_1.svg")?;
+///     Ok(())
+/// }
+/// ```
+/// 
+/// ![doc_boxplot_1.svg](https://raw.githubusercontent.com/cpmech/plotpy/main/figures/doc_boxplot_1.svg)
+/// 
+/// 
+/// ## More examples
+/// 
+/// See also integration test in the **tests** directory.
+pub struct Boxplot {
+    label: String,              // Name of this box in the legend
+    symbol: Option<String>,     // The default symbol for flier (outlier) points.
+    vertical: Option<bool>,     // Vertical boxplot
```

Comment:
Hi, I also thought a lot when deciding whether to use `Option<bool>` here. If we use `Option<bool>`, we don't need to change our code when matplotlib changes it's default value, such as switching the default value from `True` to `False`. If we use `bool` instead of `Option<bool>`,  the boxplot will be horizontal even if when we set vertical to `true`, in case matplotlib changes it's default value.

Reply:
Thanks again. I prefer to have `horizontal` as in Barplot and Histogram. We can be different than Matplotlib (as explained in the main Readme). The extra functionality is something exceptional that should be reduced as we implemented more functions.